src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCacheOptions.cs

```diff
@@ -19,20 +19,24 @@ public class MemoryCacheOptions : IOptions<MemoryCacheOptions>
         /// </summary>
         public TimeSpan ExpirationScanFrequency { get; set; } = TimeSpan.FromMinutes(1);
 
+        internal bool HasSizeLimit => _sizeLimit >= 0;
+
+        internal long SizeLimitValue => _sizeLimit;
+
         /// <summary>
         /// Gets or sets the maximum size of the cache.
         /// </summary>
         public long? SizeLimit
         {
-            get => _sizeLimit;
+            get => _sizeLimit < 0 ? null : _sizeLimit;
             set
             {
                 if (value < 0)
                 {
                     throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} must be non-negative.");
                 }
 
-                _sizeLimit = value;
+                _sizeLimit = value ?? -1;
```

Comment:
It doesn't really matter if it's -1, could use any negative value. This would be the only place where that constant is used, I'm not sure what it would even be named.

Reply:
+1, as it would make the code more readable.